File: WordApp/Models/UserModel.cs
using Entities.Enums;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using WordApp.Models.Base;

namespace WordApp.Models
{
    public class UserModel: BaseModel
    {
        public string Name { get; set; }
        public string Password { get; set; }

        [JsonConverter(typeof(StringEnumConverter))]
        public UserType UserType { get; set; }
        public string Token { get; set; }
    }
}

File: WordApp/ClientApp/src/app/models/users/user.model.ts
import { EUserType } from './../../app-enums';
import { UserTokenModel } from './user-token.model';

export class UserModel {
  name: string;
  password: string;
  userType: EUserType;
  id: string;
  rowVersion: string;
  token: UserTokenModel;
  email: string;

  constructor(name: string, password: string, userType: EUserType, email: string, id: string, rowVersion: string) {
    this.name = name;
    this.password = password;
    this.userType = userType;
    this.id = id;
    this.rowVersion = rowVersion;
    this.email = email;
  }
}
